zebra-consensus/src/chain.rs

```diff
@@ -79,33 +90,82 @@ where
         let mut state_service = self.state_service.clone();
         let max_checkpoint_height = self.max_checkpoint_height;
 
+        let height = block.coinbase_height();
+        // Report each 1000th block at info level
+        let info_log = matches!(height, Some(BlockHeight(height)) if (height % 1000 == 0));
+
+        if info_log {
+            tracing::info!(?height, "ChainVerifier received block");
+        } else {
+            tracing::debug!(?height, "ChainVerifier received block");
+        }
+
+        // Log a warning on unexpected high blocks
+        let is_unexpected_high_block = match height {
+            Some(BlockHeight(height))
+                if (height > self.last_block_height.0 + MAX_EXPECTED_BLOCK_GAP) =>
+            {
+                true
+            }
+            Some(height) => {
+                // Update the last height if the block height was expected
+                self.last_block_height = height;
+                false
+            }
+            _ => false,
+        };
+
         async move {
-            // Call a verifier based on the block height and checkpoints
-            //
             // TODO(teor): for post-sapling checkpoint blocks, allow callers
             //             to use BlockVerifier, CheckpointVerifier, or both.
-            match block.coinbase_height() {
+
+            // Call a verifier based on the block height and checkpoints.
+            let hash = match height {
                 Some(height) if (height <= max_checkpoint_height) => {
+                    if info_log {
+                        tracing::info!(?height, "sending block to CheckpointVerifier");
+                    } else {
+                        tracing::debug!(?height, "sending block to CheckpointVerifier");
+                    }
```

Comment:
Rather than making a tracing event here, there can be a tracing event in `checkpoint_verifier.call`, which gives the same information with less redundancy.

Reply:
That's only true if `ready_and().await?` completes successfully, which was one of the things I was trying to diagnose.